storage/remote/write_handler.go

```diff
@@ -513,6 +513,7 @@ func (h *otlpWriteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	converter := otlptranslator.NewPrometheusConverter()
 	annots, err := converter.FromMetrics(r.Context(), req.Metrics(), otlptranslator.Settings{
 		AddMetricSuffixes:         true,
+		AllowUTF8:                 (model.NameValidationScheme == model.UTF8Validation) && (otlpCfg.TranslationStrategy == config.NoUTF8EscapingWithSuffixes),
```

Comment:
At this point, the name validation scheme should already be validated, and we shouldn't swallow a mistake in any case. If you want to make sure a programming error wasn't made, you should instead validate and return an error if the name validation scheme and translation strategy don't correspond to each other.
```suggestion
		AllowUTF8:                 otlpCfg.TranslationStrategy == config.NoUTF8EscapingWithSuffixes,
```